test/groovy/TransportRequestGetTransportsTest.groovy

```diff
@@ -0,0 +1,87 @@
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.junit.rules.RuleChain
+
+import util.BasePiperTest
+import util.JenkinsStepRule
+import util.JenkinsLoggingRule
+import util.Rules
+
+import hudson.AbortException
+
+
+public class TransportRequestGetTransportsTest extends BasePiperTest {
+
+    private ExpectedException thrown = new ExpectedException()
+    private JenkinsStepRule jsr = new JenkinsStepRule(this)
+    private JenkinsLoggingRule jlr = new JenkinsLoggingRule(this)
+
+    @Rule
+    public RuleChain ruleChain = Rules.getCommonRules(this)
+        .around(thrown)
+        .around(jsr)
+        .around(jlr)
+
+    @Before
+    public void setup() {
+
+        helper.registerAllowedMethod('usernamePassword', [Map.class], {m -> return m})
+
+        helper.registerAllowedMethod('withCredentials', [List, Closure], { l, c ->
+
+            credentialsId = l[0].credentialsId
+            binding.setProperty('username', 'anonymous')
+            binding.setProperty('password', '********')
+            try {
+                c()
+            } finally {
+                binding.setProperty('username', null)
+                binding.setProperty('password', null)
+            }
+         })
+
+        helper.registerAllowedMethod('sh', [Map], { Map m -> return 0 })
```

Comment:
What about using JenkinsShellCallRule?

Reply:
So far I did it like this. I can change it in another PR for all the steps later on.